kinesis/src/main/scala/com.snowplowanalytics.snowplow.collectors.scalastream/sinks/KinesisSink.scala

```diff
@@ -206,6 +209,40 @@ object KinesisSink {
       case None => None.asRight
     }
 
+  /**
+    * Check whether an SQS queue (buffer) exists
+    * @param name Name of the queue
+    * @return Whether the queue exists
+    */
+  private def sqsQueueExists(client: AmazonSQS, name: String): Boolean =
+    try {
+      client.getQueueUrl(name)
+      true
+    } catch {
+      case _: QueueDoesNotExistException => false
+    }
+
+  private def runChecks(
+    kinesisClient: AmazonKinesis,
+    streamName: String,
+    sqs: Option[SqsClientAndName]
+  ): Unit = {
+    lazy val log = LoggerFactory.getLogger(getClass())
+    val kExists  = streamExists(kinesisClient, streamName)
+
+    if (!kExists) log.error(s"Kinesis stream $streamName doesn't exist or isn't available.")
+
+    // format: off
+    sqs match {
+      case Some(clientAndName) =>
+        if (sqsQueueExists(clientAndName.sqsClient, clientAndName.sqsBufferName)) ()
+        else log.error(s"SQS queue ${clientAndName.sqsBufferName} is defined in config file, but does not exist or isn't available.")
+      case None =>
+        if (kExists) ()
+        else log.error(s"SQS buffer is not configured.")
```

Comment:
I think that it should be a `warn` rather than `error`, because it can be expected behavior that SQS is not used (it will mostly be the case)

Reply:
Ah, but we only error if there's also no Kinesis.